src/main/scala/github/gphat/censorinus/Client.scala

```diff
@@ -115,11 +147,90 @@ class Client(
   }
 
   // Encode and send a metric to something approximating statsd.
-  private def send(metric: Metric): Unit =
-    encoder.encode(metric) match {
-      case Some(message) =>
-        sender.send(message)
-      case None =>
-        log.warning(s"Unable to send metric: unsupported metric type `${metric}`")
+  private def send(metrics: Iterator[Metric]): Unit = {
+    val lines = metrics.flatMap { metric =>
+      encoder.encode(metric) match {
+        case Some(line) => Iterator.single(line)
+        case None =>
+          log.warning(s"Unable to send metric: unsupported metric type `${metric}`")
+          Iterator.empty
+      }
+    }
+
+    batcher.batch(lines) { message =>
+      sender.send(message)
+    }
+  }
+}
+
+object Client {
+
+  /**
+   * Batches metrics encoded with a [[MetricEncoder]] into byte buffers to be
+   * sent with a [[MetricSender]].
+   */
+  sealed trait Batcher {
+
+    /**
+     * Iterate over all lines, batching metrics as possible, and sending them
+     * to `f`. The `ByteBuffer` provided to `f` is only owned by the caller for
+     * the duration of `f`. Any use of the `ByteBuffer` outside of this will
+     * result in undefined and, likely, confusing/incorrect behaviour. This
+     * will completely exhaust `line`.
+     *
+     * @param lines the encoded metrics to batch
+     * @param f the caller-supplied function to send the batched metrics to
+     */
+    def batch(lines: Iterator[String])(f: ByteBuffer => Unit): Unit = {
+      val buffered = lines.buffered
+      while (lines.hasNext) {
```

Comment:
why `while(lines.hasNext) {` and not `buffered.hasNext`?

Reply:
Definitely a typo!